controllers/rabbitmqsecretengineconfig_controller.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/redhat-cop/operator-utils/pkg/util"
+	redhatcopv1alpha1 "github.com/redhat-cop/vault-config-operator/api/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	vaultutils "github.com/redhat-cop/vault-config-operator/api/v1alpha1/utils"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// RabbitMQSecretEngineConfigReconciler reconciles a RabbitMQSecretEngineConfig object
+type RabbitMQSecretEngineConfigReconciler struct {
+	util.ReconcilerBase
+	Log            logr.Logger
+	ControllerName string
+}
+
+//+kubebuilder:rbac:groups=redhatcop.redhat.io,resources=rabbitmqsecretengineconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=redhatcop.redhat.io,resources=rabbitmqsecretengineconfigs/status,verbs=get;update;patch
```

Comment:
events seems to be missing here, possibly something else like secrets....

Reply:
Secrets are listed by other controllers, do you still want me to add them for readability reasons?

Reply:
yes please

Reply:
Done